src/main/java/com/estudo/mentoria/api/controller/CategoriaController.java

```diff
@@ -0,0 +1,53 @@
+package com.estudo.mentoria.api.controller;
+
+import com.estudo.mentoria.api.mapper.CategoriaMapper;
+import com.estudo.mentoria.api.request.CategoriaRequest;
+import com.estudo.mentoria.api.response.CategoriaResponse;
+import com.estudo.mentoria.domain.services.serviceImpl.CategoriaServiceImpl;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/categorias")
+@RequiredArgsConstructor
+public class CategoriaController {
+
+    private final CategoriaServiceImpl service;
+    private final CategoriaMapper mapper;
+
+    @GetMapping
+    public ResponseEntity<Page<CategoriaResponse>> listarTodasCategorias(Pageable pageable){
+        return ResponseEntity.ok(service.findAll(pageable).map(mapper::toCategoriaResponse));
```

Comment:
Será que faz sentido ter a controller somente como um "proxy"?
Será que a lógica da service não poderia estar aqui?

- Controller "anêmica";

Reply:
Entendi e pensei a respeito, confesso que fui levado a um viés nos tutoriais que venho seguindo que a lógica do negócio ficaria apenas na service, segregando assim a responsabilidades, entretanto, com a alteração da interface fui forçado a jogar "toda a lógica de mapeamento" para a controller não a deixando tão anêmica, mas pretendo analisar mais códigos para enriquecer mais meu repertório no assunto, obrigado pela provocação.